src/algorithms/acquisition/adapters/beidou_b1cd_pcps_acquisition.cc

```diff
@@ -0,0 +1,332 @@
+ *
+ * -------------------------------------------------------------------------
+ */
+
+#include "beidou_b1cd_pcps_acquisition.h"
+#include "Beidou_B1C.h"
+#include "acq_conf.h"
+#include "beidou_b1c_signal_processing.h"
+#include "configuration_interface.h"
+#include "gnss_sdr_flags.h"
+#include <boost/math/distributions/exponential.hpp>
+#include <glog/logging.h>
+
+
+BeidouB1CdPcpsAcquisition::BeidouB1CdPcpsAcquisition(
+    ConfigurationInterface* configuration, std::string role,
+    unsigned int in_streams, unsigned int out_streams) : role_(role), in_streams_(in_streams), out_streams_(out_streams)
+{
+    configuration_ = configuration;
+    std::string default_item_type = "gr_complex";
+    std::string default_dump_filename = "./data/acquisition.dat";
+
+    LOG(INFO) << "role " << role;
+
+    item_type_ = configuration_->property(role + ".item_type", default_item_type);
+    int64_t fs_in_deprecated = configuration_->property("GNSS-SDR.internal_fs_hz", 25000000);
+    fs_in_ = configuration_->property("GNSS-SDR.internal_fs_sps", fs_in_deprecated);
+    acq_parameters_.fs_in = fs_in_;
+    dump_ = configuration_->property(role + ".dump", false);
+    acq_parameters_.dump = dump_;
+    blocking_ = configuration_->property(role + ".blocking", true);
+    acq_parameters_.blocking = blocking_;
+    doppler_max_ = configuration->property(role + ".doppler_max", 5000);
+    if (FLAGS_doppler_max != 0)
+        {
+            doppler_max_ = FLAGS_doppler_max;
+        }
+    acq_parameters_.doppler_max = doppler_max_;
+    sampled_ms_ = configuration_->property(role + ".coherent_integration_time_ms", 10);
+    acq_parameters_.sampled_ms = sampled_ms_;
+    bit_transition_flag_ = configuration_->property(role + ".bit_transition_flag", false);
+    acq_parameters_.bit_transition_flag = bit_transition_flag_;
+    use_CFAR_algorithm_flag_ = configuration_->property(role + ".use_CFAR_algorithm", true);  //will be false in future versions
+    acq_parameters_.use_CFAR_algorithm_flag = use_CFAR_algorithm_flag_;
+    max_dwells_ = configuration_->property(role + ".max_dwells", 1);
+    acq_parameters_.max_dwells = max_dwells_;
+    dump_filename_ = configuration_->property(role + ".dump_filename", default_dump_filename);
+    acq_parameters_.dump_filename = dump_filename_;
+    //--- Find number of samples per spreading code -------------------------
+    code_length_ = static_cast<uint32_t>(std::round(static_cast<double>(fs_in_) / (BEIDOU_B1Cd_CODE_RATE_HZ / static_cast<double>(BEIDOU_B1Cd_CODE_LENGTH_CHIPS))));
+
+    vector_length_ = code_length_ * sampled_ms_;
+
+    if (bit_transition_flag_)
+        {
+            vector_length_ *= 2;
+        }
+
+    code_ = new gr_complex[vector_length_];
+
+    if (item_type_.compare("cshort") == 0)
+        {
+            item_size_ = sizeof(lv_16sc_t);
+        }
+    else
+        {
+            item_size_ = sizeof(gr_complex);
+        }
+
+    acq_parameters_.ms_per_code = 1;
```

Comment:
Hey Damian, Do we need to change this value to "10"? as code period in Beidou B1C is 10ms